auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,118 @@
+        return self.with_context(auditlog_write=True).env["ir.rule"].search(
+            [("auditlog_id", "in", self.ids)])
+        
+    def get_field_ids_domain(self):
+        """note this solution will work only with a hardcoded design of models,
+        because on initialization , self.model_id.id still is not defined.
+        for now, to keep generality we put the filtering in the view."""
+        return [
+            ("model_id", "=", self.env.ref("base.model_res_partner").id),
+            ("name", "not in", FIELDS_BLACKLIST),
+        ]
+
+    def unlink(self):
+        # if we delete auditlog rule, corresponding ir.rules are removed
+        # TODO PROPOSAL: a warning here with detailed information?
+        to_delete = self.get_linked_rules()
+        res = super(AuditlogRule, self).unlink()
+        if res:
+            res = res and to_delete.unlink()
+        return res
+
+    def add_default_group_if_needed(self):
+        self.ensure_one()
+        res = False
+        if not self.group_ids and self.field_ids:
+            # if group has been removed and no group specified ad base user default
+            res = self.with_context(no_iter=True).write(
+                {"group_ids": [(6, 0, [self.env.ref("base.group_user").id])]})
+        return res
+
+    @api.model
+    def create(self, vals):
+        res = super(AuditlogRule, self).create(vals)
+        res.add_default_group_if_needed()
+        if res.needs_rule():
+            res.generate_rules()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(AuditlogRule, self).write(vals)
+        for this in self: 
+            added = this.add_default_group_if_needed()
+            if any([x in vals for x in ("group_ids", "field_ids", "model_id", "all_fields")]) or added:
+                if this.needs_rule():
+                    this.generate_rules()
+                else:
+                    this.get_linked_rules().unlink()
+        return res
+
+    def generate_rules(self):
+        old_rule = self.env["ir.rule"].search([("auditlog_id", "=", self.id)], limit=1)
+        values = self._prepare_rule_values()
+        if old_rule:
+            old_rule.with_context(auditlog_write=True).write(values)
+        else:
+            self.with_context(auditlog_write=True).env["ir.rule"].create(values)
+
+    def _prepare_rule_values(self):
+        domain_force = "[" + " ('log_id.model_id' , '=', %s)," % (self.model_id.id)
+        if self.field_ids:
+            domain_force += "('field_id', 'in',  %s)" % (self.field_ids.ids)
+        domain_force += "]"
+        return {
+            "name": "auditlog_extended_%s" % self.id,
+            "model_id": self.env.ref("auditlog.model_auditlog_log_line").id,
+            "groups": [(6, 0, self.group_ids.ids)],
+            "perm_read": True,
+            "domain_force": domain_force,
+            "auditlog_id": self.id,
+        }
+
+    @api.multi
+    def subscribe(self):
```

Comment:
Hey.... we should call `super()` here, and leave the creation of the `View logs` action of the parent module intact. Right? Removing that functionality is not "inherited module boy scout behaviour".

ON TOP OF THAT we can add our "View log lines" action, which is the added feature of our particular module.

Reply:
Ah! yes , the original is written in a way to make it un-inheritable. 
It creates  "old" action
and does not return the created action, just true.

In order to preserve inheritance i should 

               call super
               find the action it created and modify the domain with what we want.
             
will do that ( we just have to modify all actions of the rule after calling super) and we preserve inheritance chain.

Reply:
@gfcapalbo I'm not sure if I agree with this for OCA. For customer it's fine but this is destroying the 'View Logs' functionality which people may still want to have. I think we should make the "View Logs" available for admins, and the "View Log Lines" for any user that is member of a group that is allowed to see. So an admin could see both buttons, but a user only one.